### PR TITLE
Implemented screen with images in mosaic style

### PR DESCRIPTION
**Pull request by vshtuka**

### CODE REVIEW

File: AndroidProject/app/.gitignore [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Built application files
```
Comment (FAQEnD) [inline AndroidProject/app/.gitignore]: This file should be placed in a root folder of your repository

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java [hunk 2]
```diff
@@ -2,13 +2,56 @@
...
+    private RecyclerView recyclerView;
+    private RecyclerView.LayoutManager layoutManager;
+    private RecyclerView.Adapter adapter;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        recyclerView = findViewById(R.id.recyclerView);
+        layoutManager = new GridLayoutManager(this, 2, GridLayoutManager.VERTICAL, false);
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: Please avoid usage of magic numbers, like a "2" in this case. Constant is always better to read/understand

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java [hunk 3]
```diff
@@ -2,13 +2,56 @@
...
+
+        ((GridLayoutManager) layoutManager).setSpanSizeLookup(
+                new GridLayoutManager.SpanSizeLookup() {
+                    @Override
+                    public int getSpanSize(int position) {
+                        return (position % 5 == 0 ? 2 : 1);
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: Please replace this magic numbers also

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java [hunk 4]
```diff
@@ -2,13 +2,56 @@
...
+            adapter = new RecyclerAdapter(loadImagesFromAssets());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        recyclerView.setAdapter(adapter);
     }
+
+    private Drawable[] loadImagesFromAssets() throws IOException {
+        String[] fileNames = getAssets().list("MyImages");
+        assert fileNames != null;
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: "assert" keyword used only for unit testing, please never use it in development. You can replace it with "IF" keyword

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java [hunk 5]
```diff
@@ -2,13 +2,56 @@
...
+        Drawable[] drawables = new Drawable[fileNames.length];
+        for (int i = 0; i < fileNames.length; i++) {
+            InputStream inputStream = getAssets().open("MyImages/" + fileNames[i]);
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: "MyImages/" should be a const value

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java [hunk 6]
```diff
@@ -2,13 +2,56 @@
...
+            Drawable drawable = Drawable.createFromStream(inputStream, null);
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: You can inline this part to `drawables[i] = Drawable.createFromStream(inputStream, null);`

File: AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ImageViewHolder> {
+
+    private Drawable[] drawables;
+
+    RecyclerAdapter(Drawable[] drawables) {
+        this.drawables = drawables;
+    }
+
+    static class ImageViewHolder extends RecyclerView.ViewHolder {
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java]: Please add private keyword

File: AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            photo = itemView.findViewById(R.id.photo);
+        }
+    }
+
+    @Override
+    public ImageViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.album_layout, parent, false);
+        ImageViewHolder imageViewHolder = new ImageViewHolder(view);
+
+        return imageViewHolder;
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java]: Can be inline `return new ImageViewHolder(view);`

File: AndroidProject/app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1,3 +1,3 @@
 <resources>
-    <string name="app_name">Android Project</string>
+    <string name="app_name">AndroidProject</string>
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/res/values/strings.xml]: Please name app according to a repository name

**Review by FAQEnD — CHANGES_REQUESTED** [review]

File: AndroidProject/.gitignore [hunk 10]
```diff
@@ -0,0 +1,14 @@
+*.iml
```
Comment (FAQEnD) [inline AndroidProject/.gitignore]: We need only 1 .gitignore file. So this should be removed

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java [hunk 11]
```diff
@@ -2,13 +2,65 @@
...
+        layoutManager = new GridLayoutManager(this, GRID_SPAN_COUNT,
+                GridLayoutManager.VERTICAL, false);
+
+        ((GridLayoutManager) layoutManager).setSpanSizeLookup(
+                new GridLayoutManager.SpanSizeLookup() {
+                    @Override
+                    public int getSpanSize(int position) {
+                        return (position % FULL_SCREEN_WIDTH_IMAGE_POSITION == 0 ?
+                                FULL_SCREEN_WIDTH_IMAGE_GRID_SPAN_COUNT :
+                                HALF_SCREEN_WIDTH_IMAGE_GRID_SPAN_COUNT);
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: Please move this to a new method

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java [hunk 12]
```diff
@@ -2,13 +2,65 @@
...
+                    }
+                });
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(layoutManager);
+        try {
+            adapter = new RecyclerAdapter(loadImagesFromAssets());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        recyclerView.setAdapter(adapter);
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: Please move all code related to a recycler view init to a new method

File: AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ImageViewHolder> {
+
+    private Drawable[] drawables;
+
+    RecyclerAdapter(Drawable[] drawables) {
```
Comment (FAQEnD) [inline AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java]: please add public keyword

**Review by FAQEnD — CHANGES_REQUESTED** [review]

File: AndroidProject/app/.gitignore (hunk 1 above)
Comment (vshtuka) [reply AndroidProject/app/.gitignore]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java (hunk 2 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java (hunk 3 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java (hunk 4 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java (hunk 5 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java (hunk 7 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java (hunk 6 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java (hunk 8 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java]: done

File: AndroidProject/app/src/main/res/values/strings.xml (hunk 9 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/res/values/strings.xml]: done

File: AndroidProject/.gitignore (hunk 10 above)
Comment (vshtuka) [reply AndroidProject/.gitignore]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java (hunk 11 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java (hunk 12 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/MainActivity.java]: done

File: AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java (hunk 13 above)
Comment (vshtuka) [reply AndroidProject/app/src/main/java/com/example/androidproject/RecyclerAdapter.java]: done

**Review by FAQEnD — APPROVED** [review]

**Pull request merged**